corehq/apps/app_manager/views/download.py

```diff
@@ -275,11 +278,13 @@ def create_build_files_if_necessary_handling_conflicts(is_retry=False):
         try:
             payload = request.app.fetch_attachment(full_path)
         except ResourceNotFound:
-            if build_profile in request.app.build_profiles and build_profile_access:
+            if not build_profile:
```

Comment:
It would be a good deed to update the comment above ("lazily create..."). My read of this code is that we're attempting to grab the file from the app's attachments and if it's not there, it's because 1) this app is skipping creating default files on build or 2) this app uses language profiles, which aren't created until the first download attempt of a given profile, or 3) there was an unexpected error.

Reply:
https://github.com/dimagi/commcare-hq/pull/24751/commits/92845aa4f23bba841b8b6f7fd16fbbcee63ece14

Do you think if there was an unexpected error during the initial build? I dont think that would be the right thing to document, added for the other 2

Reply:
Are you referring to the `raise` later that causes this method `download_file` to be called again?
In my opinion,
Would not it be better that the comment should be present where the except block is? like
the comment there is present for legacy stuff. The following things in the method look like expected things and then fallbacks to them instead of errors. So a statement where those are happening sharing their purpose would be more effective?

Reply:
I mean the `raise` a few lines down, on line 286 in the new code. I don't feel strongly about exactly where the comments are, I just generally think the build process is under-commented and under-documented.